core/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -120,7 +122,7 @@ public MapperService(IndexSettings indexSettings, AnalysisService analysisServic
         this.searchQuoteAnalyzer = new MapperAnalyzerWrapper(analysisService.defaultSearchQuoteAnalyzer(), p -> p.searchQuoteAnalyzer());
         this.mapperRegistry = mapperRegistry;
 
-        this.dynamic = this.indexSettings.getSettings().getAsBoolean("index.mapper.dynamic", true);
+        this.dynamic = this.indexSettings.getSettings().getAsBoolean(DYNAMIC_MAPPING_ENABLED_SETTING, true);
```

Comment:
This should be using the default value of the setting, instead of an explicit `true`.